### PR TITLE
store type in descriptor serialization data

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

File: deepmd/dpmodel/descriptor/se_e2_a.py [hunk 1]
```diff
@@ -313,6 +314,8 @@ def call(
     def serialize(self) -> dict:
         """Serialize the descriptor to dict."""
         return {
+            "@class": "Descriptor",
```
Comment (njzjz) [inline deepmd/dpmodel/descriptor/se_e2_a.py:317]: I also introduced `@class` to make it more self-explanatory.

**Comment by njzjz** [comment]
Something wrong in the CI:
```
  Importing GPG key 0xC21AD6EA:
   Userid     : "AlmaLinux <packager@almalinux.org>"
   Fingerprint: E53C F5EF 91CE B0AD 1812 ECB8 51D6 647E C21A D6EA
   From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Key imported successfully
  Import of key(s) didn't help, wrong key(s)?
  Public key for glibc-gconv-extra-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-gconv-extra-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-common-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-common-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-devel-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-devel-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-headers-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-headers-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-langpack-en-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-langpack-en-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-locale-source-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-locale-source-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for glibc-minimal-langpack-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: glibc-minimal-langpack-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  Public key for libnsl-2.28-236.el8_9.12.x86_64.rpm is not installed. Failing package is: libnsl-2.28-236.el8_9.12.x86_64
   GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-AlmaLinux
  The downloaded packages were saved in cache until the next successful transaction.
  You can remove cached packages by executing 'yum clean packages'.
  Error: GPG check FAILED
                                                             ✕ 19.99s
```

File: deepmd/dpmodel/descriptor/base_descriptor.py [hunk 2]
```diff
@@ -1,8 +1,78 @@
...
+    Plugin,
+)
+
 from .make_base_descriptor import (
     make_base_descriptor,
 )
 
-BaseDescriptor = make_base_descriptor(np.ndarray, "call")
+
+class BaseDescriptor(make_base_descriptor(np.ndarray, "call")):
```
Comment (wanghan-iapcm) [inline deepmd/dpmodel/descriptor/base_descriptor.py:18]: would it be better to move the plugin support to `make_base_descriptor`?

Comment (njzjz) [reply deepmd/dpmodel/descriptor/base_descriptor.py:18]: I will try and see if there is anything wrong.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**